src/sentry/event_manager.py

```diff
@@ -1136,8 +1137,45 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
                 event.event_id,
             )
 
+        self._handle_owner_assignment(project, group, event)
+
         return group, is_new, is_regression, is_sample
 
+    def _handle_owner_assignment(self, project, group, event):
+        # Is the issue already assigned to a team or user?
+        if group.assignee_set.exists():
+            return
+
+        owner = self._find_owner(event.data, project)
+
+        if owner is not None:
+            GroupAssignee.objects.assign(group, owner)
+
+    @staticmethod
+    def _find_owner(event_data, project):
+        # Do ownership rules exist for this project?
+        if not project.projectownership_set.exists():
+            return
+
+        ownership = project.projectownership_set.first()
```

Comment:
Don't use `.first()`.

Reply:
For context we try to avoid using .first() because it forces an order. It might be ok in this situation though as I think the result set is small.

If we're going to uniformly forbid it we should just do it at the queryset level.

Reply:
I thought about this too, but did not consider it an issue, because this table should be particularly small since it is bound by the number of projects, and the following query should always return a single row (unless I am mistaken):

```sql
EXPLAIN SELECT * FROM sentry_projectownership WHERE project_id = 2 ORDER BY id;
```

I can add an explicit ORDER BY clause: `order_by('project_id')` so that `.first()` does not attempt to sort with the primary key. Using the indexed column for sorting should be always optimal either way.